Improve readme to build on linux

Hi I used the older version 2.1.0 and was able to build it with this in my dockerfile:
```
RUN apt update && apt install -y -qq --no-install-recommends build-essential pkg-config libssl-dev libcurl4-openssl-dev wget cmake && \
  wget --no-check-certificate https://github.com/mtrojnar/osslsigncode/releases/download/2.5/osslsigncode-2.5.tar.gz && \
  tar -xf osslsigncode-*.tar.gz && \
  rm *.tar.gz && \
  cd osslsigncode* && \
  ./configure && make && make install
  ```
  
According to the README.md I need to:
- Navigate to the build directory and run CMake to configure the osslsigncode project and generate a native build system:
```
mkdir build && cd build && cmake -S ..
```
with specific compile options:
```
  -DCMAKE_BUILD_TYPE=Debug
  -DCMAKE_C_COMPILER=clang
  -DCMAKE_PREFIX_PATH=[openssl directory];[curl directory]
  -DCMAKE_INSTALL_PREFIX=[installation directory]
  -DBASH_COMPLETION_USER_DIR=[bash completion installation directory]
```

When I download the tar.gz there is no "build" directory in that tar. Then I create the folder with `mkdir build` and `cd build` into that empty folder.
Then I dont know what I need for the cmake -S <options> command.
I tried this:
```
cmake -S=../ -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang 
```
With that output:
```
-- The C compiler identification is unknown
CMake Error at CMakeLists.txt:12 (project):
  The CMAKE_C_COMPILER:

    clang

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/osslsigncode-2.5/build/CMakeFiles/CMakeOutput.log".
See also "/osslsigncode-2.5/build/CMakeFiles/CMakeError.log".
```

Next I installed `apt install --no-install-recommends clang-15`

Tried the build again:
`cmake -S=../ -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang-15`
and build
`cmake --build .`

Then I got a working binary =)

--- a/README.md
+++ b/README.md
@@ -54,7 +54,7 @@ You may need to use `cmake3` instead of `cmake` to complete the following steps 
 ```
   mkdir build && cd build && cmake -S ..
 ```
-  with specific compile options:
+  optional CMake parameters:
 ```
   -DCMAKE_BUILD_TYPE=Debug
   -DCMAKE_C_COMPILER=clang

